routing/routing_integration_tests/bicycle_route_test.cpp

```diff
@@ -23,32 +25,46 @@ UNIT_TEST(RussiaMoscowSevTushinoParkPreferingBicycleWay)
 
 UNIT_TEST(RussiaMoscowNahimovskyLongRoute)
 {
+  // Get onto a secondary and follow it. Same as GraphHopper.
   CalculateRouteAndTestRouteLength(
       GetVehicleComponents(VehicleType::Bicycle),
       mercator::FromLatLon(55.66151, 37.63320), {0., 0.},
       mercator::FromLatLon(55.67695, 37.56220), 5670.0);
 }
 
-UNIT_TEST(RussiaDomodedovoSteps)
+UNIT_TEST(Russia_UseSteps)
 {
+  // Use the steps as the detour is way too long.
   CalculateRouteAndTestRouteLength(
       GetVehicleComponents(VehicleType::Bicycle),
-      mercator::FromLatLon(55.44010, 37.77416), {0., 0.},
-      mercator::FromLatLon(55.43975, 37.77272), 100.0);
+      mercator::FromLatLon(55.4403114, 37.7740223), {0., 0.},
+      mercator::FromLatLon(55.439703, 37.7725059), 139.058);
+}
+
+UNIT_TEST(Italy_AvoidSteps)
+{
+  // 690m detour instead of taking a 120m shortcut via steps.
+  // Same as Valhalla. But GraphHopper prefers steps.
+  // https://github.com/organicmaps/organicmaps/issues/2253
+  CalculateRouteAndTestRouteLength(
+      GetVehicleComponents(VehicleType::Bicycle),
+      mercator::FromLatLon(42.4631, 14.21342), {0., 0.},
+      mercator::FromLatLon(42.46343, 14.2125), 687.788);
 }
 
 UNIT_TEST(SwedenStockholmCyclewayPriority)
 {
+  /// @todo(pastk): the cycleway is the shortest and the only obvious option here anyway, what's the value of this test?
```

Comment:
detects horrific regressions that may be missed by trickier tests?

Reply:
From "CyclewayPriority" it sounds like it tries to confirm that cycleway is treated as not worse than alternatives? though maybe case where cycleway is a bit longer would be better? Can find some case for that (with use_sidepath on road)

Reply:
![image](https://github.com/user-attachments/assets/2744ee3e-ac5b-4bba-add4-d254f2dcede0)

If something goes so horrific wrong to break this case, then I'm sure it'll break many other tests also :)

Reply:
do you want a replacement for "a bit longer cycleway is better than road"?

Reply:
Yes please!

Reply:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.076924%2C19.917848%3B50.074458%2C19.910502#map=17/50.075451/19.914844&layers=N

![screen06](https://github.com/user-attachments/assets/c39e14ea-b4e6-4dfc-979d-bd883d144cef)



https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.076924%2C19.917848%3B50.074480%2C19.910488#map=18/50.075936/19.914098&layers=N

![screen07](https://github.com/user-attachments/assets/ef7f3999-b43b-4b1f-8698-b7ab1d639eb5)

Reply:
more bordeline

https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.043813%2C20.016456%3B50.047522%2C20.029986

![screen08](https://github.com/user-attachments/assets/8d1bd8bc-0164-406d-8490-32e88f1993c5)


https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.043813%2C20.016456%3B50.047522%2C20.029986#map=17/50.046302/20.023109&layers=N

![screen09](https://github.com/user-attachments/assets/4ac85516-e608-43ac-8941-731702609f6b)

Reply:
this one has contraflow on service road as cycleway substitute

https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.036180%2C19.982480%3B50.040088%2C19.983348#map=17/50.038544/19.985375&layers=N

![screen10](https://github.com/user-attachments/assets/efefe4b3-cf0b-4302-90e6-04df2ebe1269)

https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.036180%2C19.982480%3B50.040088%2C19.983348#map=17/50.038172/19.982790&layers=N

![screen11](https://github.com/user-attachments/assets/31a6e9e9-09ec-4507-b914-82a8f9e54ef4)

Reply:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.041025%2C19.962335%3B50.042704%2C19.960676#map=18/50.041848/19.961461&layers=N

![screen12](https://github.com/user-attachments/assets/622bbd00-afef-4a17-8791-1ef30e8ab4ea)


https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.041025%2C19.962335%3B50.042704%2C19.960676#map=18/50.041889/19.961488&layers=N

![screen13](https://github.com/user-attachments/assets/4511a1fa-619b-4396-8322-83f110204924)

Reply:
should be enough, I think :)

I can give more examples of specific thing from Kraków

Reply:
Many thanks! I've picked 2nd and 4th cases

Reply:
Have you pushed the new code? I see older in commits? (or maybe you selected it without turning it into code, sorry for noise then)

Reply:
I didn't manage to finish adjustments yesterday, I'll push an update later today.